nerfstudio/fields/kplanes_field.py

```diff
@@ -0,0 +1,409 @@
+            assert linear_decoder_layers is not None
+            # The NN learns a basis that is used instead of spherical harmonics
+            # Input is an encoded view direction, output is weights for
+            # combining the color features into RGB
+            # This architecture is based on instant-NGP
+            self.color_basis = tcnn.Network(
+                n_input_dims=3 + self.appearance_embedding_dim,  # self.direction_encoder.n_output_dims,
+                n_output_dims=3 * self.feature_dim,
+                network_config={
+                    "otype": "FullyFusedMLP",
+                    "activation": "ReLU",
+                    "output_activation": "None",
+                    "n_neurons": 128,
+                    "n_hidden_layers": linear_decoder_layers,
+                },
+            )
+            # sigma_net just does a linear transformation on the features to get density
+            self.sigma_net = tcnn.Network(
+                n_input_dims=self.feature_dim,
+                n_output_dims=1,
+                network_config={
+                    "otype": "CutlassMLP",
+                    "activation": "None",
+                    "output_activation": "None",
+                    "n_neurons": 128,
+                    "n_hidden_layers": 0,
+                },
+            )
+        else:
+            self.sigma_net = tcnn.Network(
+                n_input_dims=self.feature_dim,
+                n_output_dims=self.geo_feat_dim + 1,
+                network_config={
+                    "otype": "FullyFusedMLP",
+                    "activation": "ReLU",
+                    "output_activation": "None",
+                    "n_neurons": 64,
+                    "n_hidden_layers": 1,
+                },
+            )
+            self.in_dim_color = (
+                self.direction_encoding.n_output_dims + self.geo_feat_dim + self.appearance_embedding_dim
+            )
+            self.color_net = tcnn.Network(
+                n_input_dims=self.in_dim_color,
+                n_output_dims=3,
+                network_config={
+                    "otype": "FullyFusedMLP",
+                    "activation": "ReLU",
+                    "output_activation": "Sigmoid",
+                    "n_neurons": 64,
+                    "n_hidden_layers": 2,
+                },
+            )
+
+    def get_density(self, ray_samples: RaySamples) -> Tuple[TensorType, TensorType]:
+        """Computes and returns the densities."""
+        grid_input = ray_samples.frustums.get_positions().view(-1, 3)
+
+        if self.spatial_distortion is not None:
+            grid_input = self.spatial_distortion(grid_input)
+            grid_input = grid_input / 2  # from [-2, 2] to [-1, 1]
+        else:
+            # Input should be in [-1, 1]
+            grid_input = (grid_input - self.aabb[0]) * (2.0 / (self.aabb[1] - self.aabb[0])) - 1.0
+
+        if self.grid_config[0]["input_coordinate_dim"] == 4:
```

Comment:
Time needs to be normalized: in nerfstudio datasets it's in [0, 1] but for interpolation it must be in [-1, 1] (do `timestamps = (timestamps * 2) - 1`)